lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -228,10 +228,12 @@ void LazyForkOp::build(OpBuilder &builder, OperationState &result,
   // Fork is control-only if it is the no-data output of a ControlMerge or a
   // StartOp
   auto *op = operand.getDefiningOp();
-  bool isControl = ((dyn_cast<ControlMergeOp>(op) || dyn_cast<StartOp>(op)) &&
-                    operand == op->getResult(0))
-                       ? true
-                       : false;
+  bool isControl = isa_and_nonnull<ControlMergeOp, StartOp>(op) &&
```

Comment:
Why not place this in a function rather than duplicating it 3 times?

Reply:
That might be a good idea, I agree. Should I add this or do we keep these changes for later?

Reply:
Just a flyby suggestion, since I saw the same expression 3 times. Up to you :-).